### DISCUSSION TITLE
I want to achieve such a subquery, how to do it

### DISCUSSION THREAD

**Discussion by edcjian** [PostgREST/postgrest#1868] [Q&A] [answered]
```
http://111.229.235.215:3000/multi_question?questionId=in.(paper_manage?select=questionId&paperId=eq.1002&questionType=eq.1)
```

**Comment by laurenceisla** [comment, marked as answer]
Looks like embedding in PostgREST won't filter the left table, for more complicated joins it's recommended to create custom VIEWS [according to the docs](https://postgrest.org/en/stable/api.html?highlight=embedding#custom-queries). After joining `multi_question` and `paper_manage` in the VIEW `multi_question_view`, for instance, your query would look like this.

``` http
/multi_question_view?paperId=eq.1002&questionType=eq.1
```

Edit:
Taking into consideration what Steve mentioned below, a VIEW may not be your best choice and using a function is a better alternative. Something like:

``` sql
CREATE FUNCTION get_multi_question(p_id int, q_type text) RETURNS TABLE (...) AS $$
  SELECT ... FROM multi_question WHERE questionId in (
    SELECT questionId 
    FROM paper_manage 
    WHERE paperId = p_id AND questionType = q_type
  );
$$ LANGUAGE SQL STABLE;
```

And your query would look like this:

``` http
GET /rpc/get_multi_question?p_id=1002&q_type=1
```

**Reply by steve-chavez** [reply]
I think what @edcjian is trying to do is more akin to

```sql
SELECT * FROM tbl WHERE id IN (SELECT other_id FROM other_tbl WHERE id = x)
```

That looks more fit for an RPC than a view - since it needs to parametrize `x`.

**Reply by laurenceisla** [reply]
Right, an RPC should be better. Edited my answer.

**Reply by wolfgangwalther** [reply]
> ```sql
> CREATE FUNCTION get_multi_question(p_id int, q_type text) RETURNS TABLE (...) AS $$
>   SELECT ... FROM multi_question WHERE questionId in (
>     SELECT questionId 
>     FROM paper_manage 
>     WHERE paperId = p_id AND questionType = q_type
>   );
> $$ LANGUAGE SQL IMMUTABLE;
> ```

Careful - this function should for sure not be `IMMUTABLE`, but `STABLE`.

**Reply by laurenceisla** [reply]
True that. I fixed the answer.
